### PR TITLE
fix(firestore): support not equals null query

### PR DESCRIPTION
**Pull request by pafry7**
### Description

Retrieving all documents with field not equal to null doesn't work. Example:
```
firestore().collection("USERS").where("address", "!=", null);
```
Returns
```
firebase.firestore().collection().where(_, _, *) 'value' is invalid. You can only perform equals comparisons on null
```
I've looked into  `FirestoreQuery.js` and only equality comparison is allowed. 
```
if (isNull(value) && !this._modifiers.isEqualOperator(opStr)){...}
```
Is there any specific reason for it? Is there anything else I need to change, that I'm not aware of ? 

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

I've run tests locally and everything passes. I have not found a test related to this problem (maybe this is something that needs to be added).

🔥
---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ cd react-native-firebase
 yarn
 yarn tests:ios:pod:install
 brew tap wix/brew
-brew install applesimutils xbeautify
+brew install applesimutils xcbeautify
```
Comment (mikehardy) [inline CONTRIBUTING.md:105]: 🤦 thanks :-)

File: packages/firestore/lib/FirestoreQuery.js [hunk 2]
```diff
@@ -437,7 +437,11 @@ export default class FirestoreQuery {
       );
     }
 
-    if (isNull(value) && !this._modifiers.isEqualOperator(opStr)) {
+    if (
+      isNull(value) &&
+      !this._modifiers.isEqualOperator(opStr) &&
+      !this._modifiers.isNotEqualOperator(opStr)
```
Comment (mikehardy) [inline packages/firestore/lib/FirestoreQuery.js:443]: This *does* in fact appear valid + correct. `!=` is supported in comparisons against null values, so this change looks good

**Review by mikehardy — CHANGES_REQUESTED** [review]
This looks great - nice catch on a corner case missed in the implementation - you ask the question if there is any other place to change - and the answer is just in exercising it in the e2e harness to validate:

- an additional test like this one to verify that it is allowed now: https://github.com/invertase/react-native-firebase/blob/da82c1036051f0518da0401de24cef24c7ac091f/packages/firestore/e2e/Query/where.e2e.js#L87-L89

Other than that, I don't think more testing is necessary, it appears the `!=` operator is already exercised pretty well here https://github.com/invertase/react-native-firebase/blob/da82c1036051f0518da0401de24cef24c7ac091f/packages/firestore/e2e/Query/where.e2e.js#L532-L589

I think that's a really trivial addition so while I normally won't hang up on community contributions on testing, given you have installed xcbeautify (guessing from your diff line there ;-) ) adding this simple adaptation of an existing test seems like it isn't onerous?

That along with the CLA signed (check the related comment on the PR) - then good to go

What do you think?

**Comment by pafry7** [comment]
@mikehardy Done 🔥

**Review by mikehardy — APPROVED** [review]
Fantastic - thanks of course just for noticing + fixing the error, and bonus thanks for a docs fix and a test addition. Great PR
